internal/pkg/service/buffer/store/watcher.go

```diff
@@ -53,8 +58,20 @@ func (w *Watcher) GetSliceID(k key.ExportKey) (*time.Time, bool) {
 	return &sliceIDTyped, true
 }
 
-func (w *Watcher) Watch(_ context.Context, _ etcd.Client) {
-	// TODO
+func (w *Watcher) Watch(ctx context.Context, logger log.Logger, client *etcd.Client) {
+	handleErrors := func(err error) {
+		logger.Error(err)
+	}
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				break
+			case slice := <-w.store.schema.Slices().Watch(ctx, client, handleErrors):
```

Comment:
Tu mas asi chybu, `ch` sa ti vytvara pri kazdej iteracii vo `for`.

Reply:
🤔 asi ne. Takže v gorutině mám nejdřív vytvořit ten kanál do nějaký proměnný (řekněme `slicesCh`) a tady bude už jen `case slice := <- slicesCh`?